backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModularContractProvider.java

```diff
@@ -131,7 +131,13 @@ public Stream<FlatrateTermId> streamModularPurchaseContractsForPurchaseOrderLine
 	public Stream<FlatrateTermId> streamModularPurchaseContractsForReceiptLine(@NonNull final InOutLineId receiptInOutLineId)
 	{
 		final I_M_InOutLine inOutLineRecord = inoutDao.getLineByIdInTrx(receiptInOutLineId);
-		return Stream.ofNullable(FlatrateTermId.ofRepoIdOrNull(inOutLineRecord.getC_Flatrate_Term_ID()));
+		final FlatrateTermId contractId = FlatrateTermId.ofRepoIdOrNull(inOutLineRecord.getC_Flatrate_Term_ID());
+
+		return Stream.ofNullable(contractId)
+				.map(flatrateBL::getById)
+				.filter(contract -> MODULAR_CONTRACT.getCode().equals(contract.getType_Conditions()))
```

Comment:
minor: for enum, pls introduce method TypeConditions:
```java
public boolean equalsByCode(@Nullable String code) { return Objects.equals(this.code, code); }
```

and then use it here.
so we will be on the safe side.